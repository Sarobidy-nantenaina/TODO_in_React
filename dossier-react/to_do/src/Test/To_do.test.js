import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from "@testing-library/react"; 
import TodoList from '../To_do';

describe('TodoList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TodoList />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with an empty input default", () => {
    const { getByPlaceholderText } = render(<TodoList />);
    const input = getByPlaceholderText("tap here the todo...");
    expect(input.value).toEqual("");
  });

  it('renders todos as an empty array by default', () => {
    const component = renderer.create(<TodoList />);
    const instance = component.getInstance();
    if (instance) {
      expect(instance.state.todos).toEqual([]);
    } else {
      throw new Error('Cannot get instance of TodoList component');
    }
  });

  it('renders done as an empty array by default', () => {
    const component = renderer.create(<TodoList />);
    const instance = component.getInstance();
    if (instance) {
      expect(instance.state.done).toEqual([]);
    } else {
      throw new Error('Cannot get instance of TodoList component');
    }
  });

  it('renders correctly with an empty input', () => {
    const component = renderer.create(<TodoList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
