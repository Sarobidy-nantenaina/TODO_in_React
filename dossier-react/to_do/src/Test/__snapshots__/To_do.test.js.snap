// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly with an empty input 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly with empty DONE arrays 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly with empty TODO and DONE arrays 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly with empty TODO array 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;

exports[`TodoList renders correctly with empty TODO arrays 1`] = `
<div
  className="container"
>
  <form
    className="form-input"
    onSubmit={[Function]}
  >
    <input
      className="input-text"
      onChange={[Function]}
      placeholder=" tap here the todo..."
      type="text"
      value=""
    />
  </form>
  <div
    className="todo-done"
  >
    <div
      className="table-todo"
    >
      <h3>
        Todo
      </h3>
      <ul />
    </div>
    <div
      className="table-done"
    >
      <h3>
        Done
      </h3>
      <ul />
    </div>
  </div>
</div>
`;
